### ISSUE TITLE
Build fails for LibTiff.NET40, LibTiff.NetStandard and LibTiff

### ISSUE THREAD

**Issue by thinking-aloud** [BitMiracle/libtiff.net#42]
Hi,
I tried to build the library, because the NuGet version of libtiff did not work with .NET Core. I got the following errors:

```
Build started 5/19/2017 3:51:46 PM.
__________________________________________________
Project "/Users/lennart/Downloads/libtiff.net-master/LibTiff.NET40/LibTiff.NET40.csproj" (build target(s)):

Target PreBuildEvent:
    cd "/Users/lennart/Downloads/libtiff.net-master/"
    update_version_info.bat "/Users/lennart/Downloads/libtiff.net-master/LibTiff.NET40/AssemblyVersion.cs"
    
    /var/folders/_0/f4ghq9nj3_g5jgrmy683ttxm0000gp/T/tmpe104ea4c4e374a4d8c0ee8e8b6944698.exec.cmd: line 3: update_version_info.bat: command not found
    /Library/Frameworks/Mono.framework/Versions/5.0.0/lib/mono/msbuild/15.0/bin/Microsoft.Common.CurrentVersion.targets(1279,5): error MSB3073: The command "cd "/Users/lennart/Downloads/libtiff.net-master/"
    /Library/Frameworks/Mono.framework/Versions/5.0.0/lib/mono/msbuild/15.0/bin/Microsoft.Common.CurrentVersion.targets(1279,5): error MSB3073: update_version_info.bat "/Users/lennart/Downloads/libtiff.net-master/LibTiff.NET40/AssemblyVersion.cs"
    /Library/Frameworks/Mono.framework/Versions/5.0.0/lib/mono/msbuild/15.0/bin/Microsoft.Common.CurrentVersion.targets(1279,5): error MSB3073: " exited with code 127.
Done building target "PreBuildEvent" in project "LibTiff.NET40.csproj" -- FAILED.

Done building project "LibTiff.NET40.csproj" -- FAILED.
__________________________________________________
Project "/Users/lennart/Downloads/libtiff.net-master/LibTiff.NetStandard/LibTiff.NetStandard.csproj" (build target(s)):

Target PreBuildEvent:
    cd "/Users/lennart/Downloads/libtiff.net-master/"
    update_version_info.bat "/Users/lennart/Downloads/libtiff.net-master/LibTiff.NetStandard/AssemblyVersion.cs"
    /var/folders/_0/f4ghq9nj3_g5jgrmy683ttxm0000gp/T/tmpe8c6f446cebe464cb07d1a800c2dd895.exec.cmd: line 3: update_version_info.bat: command not found
    /Library/Frameworks/Mono.framework/Versions/5.0.0/lib/mono/msbuild/15.0/bin/Microsoft.Common.CurrentVersion.targets(1279,5): error MSB3073: The command "cd "/Users/lennart/Downloads/libtiff.net-master/"
    /Library/Frameworks/Mono.framework/Versions/5.0.0/lib/mono/msbuild/15.0/bin/Microsoft.Common.CurrentVersion.targets(1279,5): error MSB3073: update_version_info.bat "/Users/lennart/Downloads/libtiff.net-master/LibTiff.NetStandard/AssemblyVersion.cs"" exited with code 127.
Done building target "PreBuildEvent" in project "LibTiff.NetStandard.csproj" -- FAILED.

Done building project "LibTiff.NetStandard.csproj" -- FAILED.
__________________________________________________
Project "/Users/lennart/Downloads/libtiff.net-master/LibTiff/LibTiff.csproj" (build target(s)):

Target PreBuildEvent:
    cd "/Users/lennart/Downloads/libtiff.net-master/"
    update_version_info.bat "/Users/lennart/Downloads/libtiff.net-master/LibTiff/AssemblyVersion.cs"
    
    /var/folders/_0/f4ghq9nj3_g5jgrmy683ttxm0000gp/T/tmp7cf2ee09bc3c4beeb9f5b28f322c619b.exec.cmd: line 3: update_version_info.bat: command not found
    /Library/Frameworks/Mono.framework/Versions/5.0.0/lib/mono/msbuild/15.0/bin/Microsoft.Common.CurrentVersion.targets(1279,5): error MSB3073: The command "cd "/Users/lennart/Downloads/libtiff.net-master/"
    /Library/Frameworks/Mono.framework/Versions/5.0.0/lib/mono/msbuild/15.0/bin/Microsoft.Common.CurrentVersion.targets(1279,5): error MSB3073: update_version_info.bat "/Users/lennart/Downloads/libtiff.net-master/LibTiff/AssemblyVersion.cs"
    /Library/Frameworks/Mono.framework/Versions/5.0.0/lib/mono/msbuild/15.0/bin/Microsoft.Common.CurrentVersion.targets(1279,5): error MSB3073: " exited with code 127.
Done building target "PreBuildEvent" in project "LibTiff.csproj" -- FAILED.

Done building project "LibTiff.csproj" -- FAILED.
```

**Comment by Bobrovsky** [comment]
Hello Lennart,

It looks like you try to use the library and project on a platform other than Windows. What OS do you use? What tool do you use to build the library? What tool did you use to add NuGet dependency?

**Comment by thinking-aloud** [comment]
I tried it on OSX, but eventually it has to run on Linux.
I keep going back  back and forth between Rider and VS Code, but I also tried "dotnet build" directly.

**Comment by Bobrovsky** [comment]
Lennart,

Thank you for the additional information. 

I am not sure why NuGet package does not work in your case. There is a version that should work with .NET Core just fine. Please provide more information about why it does not work in your case.

As for the build errors, they are because the projects use `cmd.exe` and `cscript` to update version history before the build. Since we never build the library in operating systems other than Windows, it's not a problem for us. But it's clearly prevents builds on Mac or Linux. 

We will be happy to receive a pull request if you decide to fix the build process on Mac / Linux.

**Comment by thinking-aloud** [comment]
I will have a closer look tomorrow, when I'm in the office.
Just one question. Am I right to assume that the Open(path, "r") allows reading a file in parallel? This is the main reason I wanted to try libtiff.

**Comment by Bobrovsky** [comment]
Open(path, "r") opens file shared for reading. In other words, subsequent opening of the file for reading is allowed. At least on Windows platform.

**Comment by thinking-aloud** [comment]
Since I think the not working NuGet package is something else I opened a new issue for it: https://github.com/BitMiracle/libtiff.net/issues/43

**Comment by Bobrovsky** [comment]
@thinking-aloud I am closing this for now. Please let me know if you still observe the issue.
